### ISSUE TITLE
Investigate Better Ways to Assign Department for New Prof

### ISSUE THREAD

**Issue by AddisonTustin** [Polyratings/polyratings#47]
When someone is adding a professor to the site, we have them select the department they're in and the dept/course combo separately. This potentially causes issues where a student reviews the correct course, but doesn't change from the default department on the new professor form.

See [this search view](https://polyratings.dev/search/name?term=hickman) but [this professor view](https://polyratings.dev/professor/27f53828-8c54-45bd-a75c-312a894061fe).

**Comment by Yash-Var** [comment]
@AddisonTustin i want to fix this issue can you share me all the preferences to related issue ?

**Comment by AddisonTustin** [comment]
Hi @Yash-Var, I'm grateful for your interest in helping on the project! This week and next week are our final examinations at my university, so my time will be limited until then. 

Additionally, the other maintainer and I need to do some requirements analyses to determine what solutions would be most convenient/useful to students.

I will most certainly be in touch

**Comment by Chris2fourlaw** [comment]
Yo @AddisonTustin so I'm thinking we can do some form validation to if we are adding a professor, we remove the dropdown on the review part and use the dept dropdown

This doesn't solve the issue where the professor swaps/teaches across departments though. I think maybe we can do some more form validation like an "are you sure" if the review is under a class that differs from the professors dept.

Even deeper, why tag the professor with a dept separate from the classes they teach? How about we label the professor by all of the class prefixes associated with their reviews and do the alert above if a user tries to add a class with a prefix that differs from the current norm

In summary, I propose we:
- Remove dept selection from professor creation
- Run a backfill that sets a professors dept as a list of their class prefixes
- When adding a review, if the user selects a class whose prefix is different from the professors dept, prompt the user for confirmation

**Comment by mfish33** [comment]
Hey, @Chris2fourlaw I did not see this comment until today when cleaning up some of the GitHub issues. I really like the idea of removing the department label as I think it can actually be more confusing. For example, searching for a professor that you think is C.S. but is actually CPE so they do not come up when that filter is applied. Every professor has a list of courses they teach so we can use those to apply filters and rank the courses on the professor's page. 

The only thing I disagree with is running a backfill to set the departments as an array. We already have that data as the courses field and adding a data dependency is not the best.
